src/OrchardCore/OrchardCore.ReCaptcha.Core/Services/ReCaptchaService.cs

```diff
@@ -65,7 +64,8 @@ public void ThisIsAHuman()
         /// <returns></returns>
         public async Task<bool> VerifyCaptchaResponseAsync(string reCaptchaResponse)
         {
-            return !string.IsNullOrWhiteSpace(reCaptchaResponse) && await _reCaptchaClient.VerifyAsync(reCaptchaResponse, _settings.SecretKey);
+            var client = _httpContextAccessor.HttpContext.RequestServices.GetRequiredService<ReCaptchaClient>();
```

Comment:
yes this is something we can do and we do it in many places to avoid things like circular references, but I am not a big fan of this way of resolving service. Just thought to share my point of view

Reply:
Yes I understand, just because it was the simplest way to fix it during my tests.

And it was already using `_httpContextAccessor` ;)